packages/drivers/odsp-driver/src/odspDocumentServiceFactory.ts

```diff
@@ -35,3 +41,62 @@ export class OdspDocumentServiceFactory
         );
     }
 }
+
+/**
+ * Factory for creating sharepoint document service with a provided snapshot.
+ * Use if you don't want to connect to any kind of external/internal storages and want to provide
+ * content directly.
+ */
+export class LocalOdspDocumentServiceFactory extends OdspDocumentServiceFactoryCore {
+    private logger: TelemetryLogger | undefined;
+
+    constructor(
+        private readonly localSnapshot: Uint8Array | string,
+    ) {
+        super(
+            async (_options) => {
+                return this.throwUnsupportedUsage("Getting storage token");
+            },
+            async (_options) => {
+                return this.throwUnsupportedUsage("Getting websocket token");
+            },
+            async () => {
+                return this.throwUnsupportedUsage("Getting SocketIO Client");
+            },
+        );
+    }
+
+    private throwUnsupportedUsage<T>(unsupportedFuncName: string): T {
+        const toThrow = new UsageError(
+            `${unsupportedFuncName} is not supported by LocalOdspDocumentServiceFactory`);
+        this.logger?.sendErrorEvent({ eventName: "UnsupportedUsage" }, toThrow);
+        throw toThrow;
+    }
+
+    public async createContainer(
+        _createNewSummary: ISummaryTree | undefined,
+        _createNewResolvedUrl: IResolvedUrl,
+        logger?: ITelemetryBaseLogger,
+        _clientIsSummarizer?: boolean,
+    ): Promise<IDocumentService> {
```

Comment:
I think you should be able to mark this function as returning `never`. Does not matter much, unless this function is called directly (on this class instance)